In [31]:
!python --version

Python 3.13.5


In [32]:
# Install necessary packages
!pip install kaggle pandas mysql-connector-python sqlalchemy

In [33]:
# Import libraries
import os
import zipfile
import pandas as pd
from urllib.parse import quote_plus # Only if your passowrd consists of '@' which can create errors for connection
from sqlalchemy import create_engine

In [34]:
# After creating the token from kaggle website in the profile settings, paste below

In [35]:
os.environ["KAGGLE_API_TOKEN"] = "PASTE YOUR TOKEN"

In [36]:
# Download Dataset from kaggle
!kaggle datasets download -d ankitbansal06/retail-orders

In [37]:
# Downloaded file from kaggle will be a zip file 
#Unzipping the downloaded file
with zipfile.ZipFile("retail-orders.zip",'r') as zipref:
    zipref.extractall() # If you want to extract in a particular location, then it is to be specified in the brackets

Exploratory Data Analysis

In [38]:
# Reading the file for understanding the data
df = pd.read_csv('orders.csv')
df

Order Id  Order Date       Ship Mode    Segment        Country  \
0            1  2023-03-01    Second Class   Consumer  United States   
1            2  2023-08-15    Second Class   Consumer  United States   
2            3  2023-01-10    Second Class  Corporate  United States   
3            4  2022-06-18  Standard Class   Consumer  United States   
4            5  2022-07-13  Standard Class   Consumer  United States   
...        ...         ...             ...        ...            ...   
9989      9990  2023-02-18    Second Class   Consumer  United States   
9990      9991  2023-03-17  Standard Class   Consumer  United States   
9991      9992  2022-08-07  Standard Class   Consumer  United States   
9992      9993  2022-11-19  Standard Class   Consumer  United States   
9993      9994  2022-07-17    Second Class   Consumer  United States   

                 City       State  Postal Code Region         Category  \
0           Henderson    Kentucky        42420  South        Furniture   
1           Henderson    Kentucky        42420  South        Furniture   
2         Los Angeles  California        90036   West  Office Supplies   
3     Fort Lauderdale     Florida        33311  South        Furniture   
4     Fort Lauderdale     Florida        33311  South  Office Supplies   
...               ...         ...          ...    ...              ...   
9989            Miami     Florida        33180  South        Furniture   
9990       Costa Mesa  California        92627   West        Furniture   
9991       Costa Mesa  California        92627   West       Technology   
9992       Costa Mesa  California        92627   West  Office Supplies   
9993      Westminster  California        92683   West  Office Supplies   

     Sub Category       Product Id  cost price  List Price  Quantity  \
0       Bookcases  FUR-BO-10001798         240         260         2   
1          Chairs  FUR-CH-10000454         600         730         3   
2          Labels  OFF-LA-10000240          10          10         2   
3          Tables  FUR-TA-10000577         780         960         5   
4         Storage  OFF-ST-10000760          20          20         2   
...           ...              ...         ...         ...       ...   
9989  Furnishings  FUR-FU-10001889          30          30         3   
9990  Furnishings  FUR-FU-10000747          70          90         2   
9991       Phones  TEC-PH-10003645         220         260         2   
9992        Paper  OFF-PA-10004041          30          30         4   
9993   Appliances  OFF-AP-10002684         210         240         2   

      Discount Percent  
0                    2  
1                    3  
2                    5  
3                    2  
4                    5  
...                ...  
9989                 4  
9990                 4  
9991                 2  
9992                 3  
9993                 3  

[9994 rows x 16 columns]

In [39]:
# To check the datatypes and no. of Null values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Order Id          9994 non-null   int64 
 1   Order Date        9994 non-null   object
 2   Ship Mode         9993 non-null   object
 3   Segment           9994 non-null   object
 4   Country           9994 non-null   object
 5   City              9994 non-null   object
 6   State             9994 non-null   object
 7   Postal Code       9994 non-null   int64 
 8   Region            9994 non-null   object
 9   Category          9994 non-null   object
 10  Sub Category      9994 non-null   object
 11  Product Id        9994 non-null   object
 12  cost price        9994 non-null   int64 
 13  List Price        9994 non-null   int64 
 14  Quantity          9994 non-null   int64 
 15  Discount Percent  9994 non-null   int64 
dtypes: int64(6), object(10)
memory usage: 1.2+ MB


In [40]:
# Basic Statistics of numerical columns
df.describe()

Order Id   Postal Code    cost price    List Price     Quantity  \
count  9994.000000   9994.000000   9994.000000   9994.000000  9994.000000   
mean   4997.500000  55190.379428    201.189714    229.756854     3.789574   
std    2885.163629  32063.693350    537.743203    623.245839     2.225110   
min       1.000000   1040.000000      0.000000      0.000000     1.000000   
25%    2499.250000  23223.000000     20.000000     20.000000     2.000000   
50%    4997.500000  56430.500000     50.000000     50.000000     3.000000   
75%    7495.750000  90008.000000    180.000000    210.000000     5.000000   
max    9994.000000  99301.000000  18110.000000  22640.000000    14.000000   

       Discount Percent  
count       9994.000000  
mean           3.484090  
std            1.114211  
min            2.000000  
25%            2.000000  
50%            3.000000  
75%            4.000000  
max            5.000000

Checking unique values and no. of unique values in the object type columns

In [41]:
df['Ship Mode'].unique()

array(['Second Class', 'Standard Class', 'Not Available', 'unknown',
       'First Class', nan, 'Same Day'], dtype=object)

Found that Ship Mode has values 'Not Available' and 'unknown', which are basically nan values, 
but python is not detecting them as nan. Will have to resolve this 

In [42]:
df['Segment'].unique()

array(['Consumer', 'Corporate', 'Home Office'], dtype=object)

In [43]:
df['City'].nunique()

531

In [44]:
df['Country'].unique()

array(['United States'], dtype=object)

In [45]:
df['State'].nunique()

49

In [46]:
df['Region'].unique()

array(['South', 'West', 'Central', 'East'], dtype=object)

In [47]:
df['Category'].unique()

array(['Furniture', 'Office Supplies', 'Technology'], dtype=object)

In [48]:
df['Sub Category'].unique()

array(['Bookcases', 'Chairs', 'Labels', 'Tables', 'Storage',
       'Furnishings', 'Art', 'Phones', 'Binders', 'Appliances', 'Paper',
       'Accessories', 'Envelopes', 'Fasteners', 'Supplies', 'Machines',
       'Copiers'], dtype=object)

In [49]:
df[df['Category']=='Furniture']['Sub Category'].unique()

array(['Bookcases', 'Chairs', 'Tables', 'Furnishings'], dtype=object)

In [50]:
df[df['Category']=='Office Supplies']['Sub Category'].unique()

array(['Labels', 'Storage', 'Art', 'Binders', 'Appliances', 'Paper',
       'Envelopes', 'Fasteners', 'Supplies'], dtype=object)

In [51]:
df[df['Category']=='Technology']['Sub Category'].unique()

array(['Phones', 'Accessories', 'Machines', 'Copiers'], dtype=object)

In [52]:
df['Product Id'].nunique()

1862

In [68]:
df[df['cost price']==0.0]

Order Id  Order Date       Ship Mode      Segment        Country  \
15          16  2022-06-18  Standard Class  Home Office  United States   
30          31  2022-11-20  Standard Class     Consumer  United States   
61          62  2023-04-08     First Class     Consumer  United States   
70          71  2022-11-29  Standard Class     Consumer  United States   
75          76  2023-04-17     First Class    Corporate  United States   
...        ...         ...             ...          ...            ...   
9838      9839  2023-03-01  Standard Class     Consumer  United States   
9859      9860  2022-08-13  Standard Class  Home Office  United States   
9869      9870  2023-01-02     First Class    Corporate  United States   
9945      9946  2023-06-26  Standard Class    Corporate  United States   
9965      9966  2022-12-31    Second Class    Corporate  United States   

               City         State  Postal Code   Region         Category  \
15       Fort Worth         Texas        76106  Central  Office Supplies   
30     Philadelphia  Pennsylvania        19140     East  Office Supplies   
61             Troy      New York        12180     East  Office Supplies   
70    New York City      New York        10009     East  Office Supplies   
75          Houston         Texas        77041  Central  Office Supplies   
...             ...           ...          ...      ...              ...   
9838     Long Beach    California        90805     West  Office Supplies   
9859       Richmond       Indiana        47374  Central  Office Supplies   
9869          Mason          Ohio        45040     East  Office Supplies   
9945   Philadelphia  Pennsylvania        19120     East  Office Supplies   
9965         Newark      Delaware        19711     East  Office Supplies   

     Sub Category       Product Id  cost price  List Price  Quantity  \
15        Binders  OFF-BI-10000756           0           0         3   
30      Envelopes  OFF-EN-10001509           0           0         2   
61            Art  OFF-AR-10001868           0           0         1   
70        Binders  OFF-BI-10004654           0           0         1   
75        Binders  OFF-BI-10004182           0           0         3   
...           ...              ...         ...         ...       ...   
9838       Labels  OFF-LA-10001771           0           0         1   
9859    Fasteners  OFF-FA-10003472           0           0         2   
9869      Binders  OFF-BI-10000494           0           0         2   
9945        Paper  OFF-PA-10000327           0           0         1   
9965        Paper  OFF-PA-10000349           0           0         1   

      Discount Percent  
15                   5  
30                   5  
61                   5  
70                   3  
75                   4  
...                ...  
9838                 3  
9859                 2  
9869                 5  
9945                 4  
9965                 5  

[507 rows x 16 columns]

Observations:
This is the sales data of a company for the years 2022 and 2023 based in USA in 49 states selling equipments used in offices .
There are 3 segments - Consumer, Corporate, Home Office
There are 3 categories - Furniture, Office Supplies, Technology
There are 17 sub categories total
There are 4 regions: East, West, South and Central
There are a total of 1862 products

Some sales are with price 0
All sales have minimum quantity 1 and discount percent 2 

Data Cleaning

In [53]:
# Reading the file again to resolve the nan issue in ship mode column and others if any
df1 = pd.read_csv('orders.csv',keep_default_na=['Not Available','unknown'])

In [54]:
# Datatype corrections: Order date to datetime(For analysis based on date) and postal code as string/object(Similar to id, calculations are not done)
df1['Order Date'] = pd.to_datetime(df1['Order Date'],format = '%Y-%m-%d')
df1['Postal Code'] = df1['Postal Code'].astype(str)
df1

Order Id Order Date       Ship Mode    Segment        Country  \
0            1 2023-03-01    Second Class   Consumer  United States   
1            2 2023-08-15    Second Class   Consumer  United States   
2            3 2023-01-10    Second Class  Corporate  United States   
3            4 2022-06-18  Standard Class   Consumer  United States   
4            5 2022-07-13  Standard Class   Consumer  United States   
...        ...        ...             ...        ...            ...   
9989      9990 2023-02-18    Second Class   Consumer  United States   
9990      9991 2023-03-17  Standard Class   Consumer  United States   
9991      9992 2022-08-07  Standard Class   Consumer  United States   
9992      9993 2022-11-19  Standard Class   Consumer  United States   
9993      9994 2022-07-17    Second Class   Consumer  United States   

                 City       State Postal Code Region         Category  \
0           Henderson    Kentucky       42420  South        Furniture   
1           Henderson    Kentucky       42420  South        Furniture   
2         Los Angeles  California       90036   West  Office Supplies   
3     Fort Lauderdale     Florida       33311  South        Furniture   
4     Fort Lauderdale     Florida       33311  South  Office Supplies   
...               ...         ...         ...    ...              ...   
9989            Miami     Florida       33180  South        Furniture   
9990       Costa Mesa  California       92627   West        Furniture   
9991       Costa Mesa  California       92627   West       Technology   
9992       Costa Mesa  California       92627   West  Office Supplies   
9993      Westminster  California       92683   West  Office Supplies   

     Sub Category       Product Id  cost price  List Price  Quantity  \
0       Bookcases  FUR-BO-10001798         240         260         2   
1          Chairs  FUR-CH-10000454         600         730         3   
2          Labels  OFF-LA-10000240          10          10         2   
3          Tables  FUR-TA-10000577         780         960         5   
4         Storage  OFF-ST-10000760          20          20         2   
...           ...              ...         ...         ...       ...   
9989  Furnishings  FUR-FU-10001889          30          30         3   
9990  Furnishings  FUR-FU-10000747          70          90         2   
9991       Phones  TEC-PH-10003645         220         260         2   
9992        Paper  OFF-PA-10004041          30          30         4   
9993   Appliances  OFF-AP-10002684         210         240         2   

      Discount Percent  
0                    2  
1                    3  
2                    5  
3                    2  
4                    5  
...                ...  
9989                 4  
9990                 4  
9991                 2  
9992                 3  
9993                 3  

[9994 rows x 16 columns]

In [55]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Order Id          9994 non-null   int64         
 1   Order Date        9994 non-null   datetime64[ns]
 2   Ship Mode         9993 non-null   object        
 3   Segment           9994 non-null   object        
 4   Country           9994 non-null   object        
 5   City              9994 non-null   object        
 6   State             9994 non-null   object        
 7   Postal Code       9994 non-null   object        
 8   Region            9994 non-null   object        
 9   Category          9994 non-null   object        
 10  Sub Category      9994 non-null   object        
 11  Product Id        9994 non-null   object        
 12  cost price        9994 non-null   int64         
 13  List Price        9994 non-null   int64         
 14  Quantity          9994 n

Making the column names lowercase and replacing the spaces with underscores as part of good practices and compatibility with SQL

In [56]:
df1.columns = df1.columns.str.lower()
df1

order id order date       ship mode    segment        country  \
0            1 2023-03-01    Second Class   Consumer  United States   
1            2 2023-08-15    Second Class   Consumer  United States   
2            3 2023-01-10    Second Class  Corporate  United States   
3            4 2022-06-18  Standard Class   Consumer  United States   
4            5 2022-07-13  Standard Class   Consumer  United States   
...        ...        ...             ...        ...            ...   
9989      9990 2023-02-18    Second Class   Consumer  United States   
9990      9991 2023-03-17  Standard Class   Consumer  United States   
9991      9992 2022-08-07  Standard Class   Consumer  United States   
9992      9993 2022-11-19  Standard Class   Consumer  United States   
9993      9994 2022-07-17    Second Class   Consumer  United States   

                 city       state postal code region         category  \
0           Henderson    Kentucky       42420  South        Furniture   
1           Henderson    Kentucky       42420  South        Furniture   
2         Los Angeles  California       90036   West  Office Supplies   
3     Fort Lauderdale     Florida       33311  South        Furniture   
4     Fort Lauderdale     Florida       33311  South  Office Supplies   
...               ...         ...         ...    ...              ...   
9989            Miami     Florida       33180  South        Furniture   
9990       Costa Mesa  California       92627   West        Furniture   
9991       Costa Mesa  California       92627   West       Technology   
9992       Costa Mesa  California       92627   West  Office Supplies   
9993      Westminster  California       92683   West  Office Supplies   

     sub category       product id  cost price  list price  quantity  \
0       Bookcases  FUR-BO-10001798         240         260         2   
1          Chairs  FUR-CH-10000454         600         730         3   
2          Labels  OFF-LA-10000240          10          10         2   
3          Tables  FUR-TA-10000577         780         960         5   
4         Storage  OFF-ST-10000760          20          20         2   
...           ...              ...         ...         ...       ...   
9989  Furnishings  FUR-FU-10001889          30          30         3   
9990  Furnishings  FUR-FU-10000747          70          90         2   
9991       Phones  TEC-PH-10003645         220         260         2   
9992        Paper  OFF-PA-10004041          30          30         4   
9993   Appliances  OFF-AP-10002684         210         240         2   

      discount percent  
0                    2  
1                    3  
2                    5  
3                    2  
4                    5  
...                ...  
9989                 4  
9990                 4  
9991                 2  
9992                 3  
9993                 3  

[9994 rows x 16 columns]

In [57]:
df1.columns = df1.columns.str.replace(' ','_')
df1

order_id order_date       ship_mode    segment        country  \
0            1 2023-03-01    Second Class   Consumer  United States   
1            2 2023-08-15    Second Class   Consumer  United States   
2            3 2023-01-10    Second Class  Corporate  United States   
3            4 2022-06-18  Standard Class   Consumer  United States   
4            5 2022-07-13  Standard Class   Consumer  United States   
...        ...        ...             ...        ...            ...   
9989      9990 2023-02-18    Second Class   Consumer  United States   
9990      9991 2023-03-17  Standard Class   Consumer  United States   
9991      9992 2022-08-07  Standard Class   Consumer  United States   
9992      9993 2022-11-19  Standard Class   Consumer  United States   
9993      9994 2022-07-17    Second Class   Consumer  United States   

                 city       state postal_code region         category  \
0           Henderson    Kentucky       42420  South        Furniture   
1           Henderson    Kentucky       42420  South        Furniture   
2         Los Angeles  California       90036   West  Office Supplies   
3     Fort Lauderdale     Florida       33311  South        Furniture   
4     Fort Lauderdale     Florida       33311  South  Office Supplies   
...               ...         ...         ...    ...              ...   
9989            Miami     Florida       33180  South        Furniture   
9990       Costa Mesa  California       92627   West        Furniture   
9991       Costa Mesa  California       92627   West       Technology   
9992       Costa Mesa  California       92627   West  Office Supplies   
9993      Westminster  California       92683   West  Office Supplies   

     sub_category       product_id  cost_price  list_price  quantity  \
0       Bookcases  FUR-BO-10001798         240         260         2   
1          Chairs  FUR-CH-10000454         600         730         3   
2          Labels  OFF-LA-10000240          10          10         2   
3          Tables  FUR-TA-10000577         780         960         5   
4         Storage  OFF-ST-10000760          20          20         2   
...           ...              ...         ...         ...       ...   
9989  Furnishings  FUR-FU-10001889          30          30         3   
9990  Furnishings  FUR-FU-10000747          70          90         2   
9991       Phones  TEC-PH-10003645         220         260         2   
9992        Paper  OFF-PA-10004041          30          30         4   
9993   Appliances  OFF-AP-10002684         210         240         2   

      discount_percent  
0                    2  
1                    3  
2                    5  
3                    2  
4                    5  
...                ...  
9989                 4  
9990                 4  
9991                 2  
9992                 3  
9993                 3  

[9994 rows x 16 columns]

In [58]:
# Creating the columns discount, sale_price and profit
df1['discount'] = df1['list_price']*df1['discount_percent']*0.01
df1['sale_price'] = df1['list_price'] - df1['discount']
df1['profit'] = df1['sale_price'] - df1['cost_price']
df1

order_id order_date       ship_mode    segment        country  \
0            1 2023-03-01    Second Class   Consumer  United States   
1            2 2023-08-15    Second Class   Consumer  United States   
2            3 2023-01-10    Second Class  Corporate  United States   
3            4 2022-06-18  Standard Class   Consumer  United States   
4            5 2022-07-13  Standard Class   Consumer  United States   
...        ...        ...             ...        ...            ...   
9989      9990 2023-02-18    Second Class   Consumer  United States   
9990      9991 2023-03-17  Standard Class   Consumer  United States   
9991      9992 2022-08-07  Standard Class   Consumer  United States   
9992      9993 2022-11-19  Standard Class   Consumer  United States   
9993      9994 2022-07-17    Second Class   Consumer  United States   

                 city       state postal_code region         category  \
0           Henderson    Kentucky       42420  South        Furniture   
1           Henderson    Kentucky       42420  South        Furniture   
2         Los Angeles  California       90036   West  Office Supplies   
3     Fort Lauderdale     Florida       33311  South        Furniture   
4     Fort Lauderdale     Florida       33311  South  Office Supplies   
...               ...         ...         ...    ...              ...   
9989            Miami     Florida       33180  South        Furniture   
9990       Costa Mesa  California       92627   West        Furniture   
9991       Costa Mesa  California       92627   West       Technology   
9992       Costa Mesa  California       92627   West  Office Supplies   
9993      Westminster  California       92683   West  Office Supplies   

     sub_category       product_id  cost_price  list_price  quantity  \
0       Bookcases  FUR-BO-10001798         240         260         2   
1          Chairs  FUR-CH-10000454         600         730         3   
2          Labels  OFF-LA-10000240          10          10         2   
3          Tables  FUR-TA-10000577         780         960         5   
4         Storage  OFF-ST-10000760          20          20         2   
...           ...              ...         ...         ...       ...   
9989  Furnishings  FUR-FU-10001889          30          30         3   
9990  Furnishings  FUR-FU-10000747          70          90         2   
9991       Phones  TEC-PH-10003645         220         260         2   
9992        Paper  OFF-PA-10004041          30          30         4   
9993   Appliances  OFF-AP-10002684         210         240         2   

      discount_percent  discount  sale_price  profit  
0                    2       5.2       254.8    14.8  
1                    3      21.9       708.1   108.1  
2                    5       0.5         9.5    -0.5  
3                    2      19.2       940.8   160.8  
4                    5       1.0        19.0    -1.0  
...                ...       ...         ...     ...  
9989                 4       1.2        28.8    -1.2  
9990                 4       3.6        86.4    16.4  
9991                 2       5.2       254.8    34.8  
9992                 3       0.9        29.1    -0.9  
9993                 3       7.2       232.8    22.8  

[9994 rows x 19 columns]

In [59]:
# Dropping the columns list_price, discount_percent and cost_price as they are no longer necessary for our analysis
df1.drop(columns = ['list_price','discount_percent','cost_price'],inplace=True)
df1

order_id order_date       ship_mode    segment        country  \
0            1 2023-03-01    Second Class   Consumer  United States   
1            2 2023-08-15    Second Class   Consumer  United States   
2            3 2023-01-10    Second Class  Corporate  United States   
3            4 2022-06-18  Standard Class   Consumer  United States   
4            5 2022-07-13  Standard Class   Consumer  United States   
...        ...        ...             ...        ...            ...   
9989      9990 2023-02-18    Second Class   Consumer  United States   
9990      9991 2023-03-17  Standard Class   Consumer  United States   
9991      9992 2022-08-07  Standard Class   Consumer  United States   
9992      9993 2022-11-19  Standard Class   Consumer  United States   
9993      9994 2022-07-17    Second Class   Consumer  United States   

                 city       state postal_code region         category  \
0           Henderson    Kentucky       42420  South        Furniture   
1           Henderson    Kentucky       42420  South        Furniture   
2         Los Angeles  California       90036   West  Office Supplies   
3     Fort Lauderdale     Florida       33311  South        Furniture   
4     Fort Lauderdale     Florida       33311  South  Office Supplies   
...               ...         ...         ...    ...              ...   
9989            Miami     Florida       33180  South        Furniture   
9990       Costa Mesa  California       92627   West        Furniture   
9991       Costa Mesa  California       92627   West       Technology   
9992       Costa Mesa  California       92627   West  Office Supplies   
9993      Westminster  California       92683   West  Office Supplies   

     sub_category       product_id  quantity  discount  sale_price  profit  
0       Bookcases  FUR-BO-10001798         2       5.2       254.8    14.8  
1          Chairs  FUR-CH-10000454         3      21.9       708.1   108.1  
2          Labels  OFF-LA-10000240         2       0.5         9.5    -0.5  
3          Tables  FUR-TA-10000577         5      19.2       940.8   160.8  
4         Storage  OFF-ST-10000760         2       1.0        19.0    -1.0  
...           ...              ...       ...       ...         ...     ...  
9989  Furnishings  FUR-FU-10001889         3       1.2        28.8    -1.2  
9990  Furnishings  FUR-FU-10000747         2       3.6        86.4    16.4  
9991       Phones  TEC-PH-10003645         2       5.2       254.8    34.8  
9992        Paper  OFF-PA-10004041         4       0.9        29.1    -0.9  
9993   Appliances  OFF-AP-10002684         2       7.2       232.8    22.8  

[9994 rows x 16 columns]

In [60]:
password = quote_plus("Your@Password") # My password consists of @. You can it directly if your do not
engine = create_engine(f"mysql+mysqlconnector://root:{password}@localhost/data_analysis") # data_analysis is the name of my db
df1.to_sql(name="daproject1",con = engine, if_exists = "fail", index=False) # here name is the table name. 
#'fail' creates a table if table doesnot exist. If table exist, raises error

In [61]:
df1.to_sql(name="daproject1",con = engine, if_exists = "replace", index=False) # Just for practice replaces the earlier table (same name) with this

In [62]:
df1.to_sql(name="daproject",con = engine, if_exists = "append", index=False) # This will add the data in the existing table